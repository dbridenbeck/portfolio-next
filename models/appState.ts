export interface ProjectModel {
  title: string;
  type: string;
  tech: string[];
  projectDetail: string;
  url: string;
  color: string;
  webm: string;
  mp4: string;
  open: boolean;
  thumb: string;
  thumbAlt: string;
}

export default interface AppStateModel {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  projects: ProjectModel[];
  pageClickedOnce: boolean;
}
