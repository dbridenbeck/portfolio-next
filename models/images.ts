export default interface ImageModel {
  currentPage: string;
  pageClickedOnce: boolean;
}