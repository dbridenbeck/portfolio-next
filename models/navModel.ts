export interface NavProps {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  changePage: (text: string) => void;
}
