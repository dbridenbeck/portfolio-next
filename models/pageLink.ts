export default interface PageLink {
  onHelloPage: boolean;
}
