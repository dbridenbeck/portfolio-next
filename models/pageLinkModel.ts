export interface PageLinkProps {
  page: { pageName: string; color: string };
  currentPage: string;
  handleChangePage: (text: string) => void;
}
