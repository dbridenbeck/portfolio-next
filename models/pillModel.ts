export interface PillProps {
  children: string;
  setMargin: string;
}
