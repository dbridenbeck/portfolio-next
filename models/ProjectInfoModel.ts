import { ProjectModel } from './appState';

export interface ProjectInfoProps {
  project: ProjectModel;
}
