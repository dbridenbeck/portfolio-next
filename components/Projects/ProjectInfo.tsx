import { motion, AnimatePresence } from "framer-motion";
import styled from "styled-components";
import { ProjectModel } from "../../models/appState";

const infoContainerVariants = {
  initial: {
    opacity: 0,
    height: "0%",
  },
  animate: {
    opacity: 1,
    height: "100%",
  },
  exit: {
    opacity: 0,
    height: "0%",
  },
};

const ProjectInfoContainer = styled(motion.div)`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  overflow: hidden;
  margin: 0;
`;

const BoldSpan = styled.span`
  font-weight: 600;
  color: ${({ color }) => color};
`;

const InfoP = styled.p`
  font-weight: normal;
  padding: 0;
  margin: 0;
`;

interface ProjectProps {
  project: ProjectModel;
  isProjectHovered: boolean;
}

const ProjectInfo: React.FC<ProjectProps> = ({
  project,
  isProjectHovered,
}) =>
  <ProjectInfoContainer
    isProjectHovered={isProjectHovered}
    initial="initial"
    exit="exit"
    animate="animate"
    variants={infoContainerVariants}
  >
    <InfoP>
      <BoldSpan color={project.color}>Goals:</BoldSpan> {project.goals}
    </InfoP>
    <InfoP>
      <BoldSpan color={project.color}>Details:</BoldSpan>{" "}
      {project.projectDetail}
    </InfoP>
  </ProjectInfoContainer>

export default ProjectInfo;
