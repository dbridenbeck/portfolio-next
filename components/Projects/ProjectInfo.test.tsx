import React from "react";
import { render } from "@testing-library/react";
import { ProjectInfoProps } from "../../models/ProjectInfoModel";
import ProjectInfo from "./ProjectInfo";

describe("ProjectInfo", () => {
  it("should render without errors", () => {
    const requiredProps: ProjectInfoProps = {
      project: {
        title: "Whidbey Herbal",
        type: "eCommerce Website",
        tech: ["React", "Redux", "Styled-Components", "Shopify API"],
        projectDetail:
          "The client's goal was to limit their reliance on Shopify by having a custom site and use images to tell their story. It was fun implementing a design file into a pixel-perfect responsive site and managing state for checkout. Roadmap is to migrate to Nextjs and add route animations.",
        url: "http://whidbeyherbal.com",
        color: "#9686ef",
        webm: `an image`,
        mp4: `an image`,
        open: false,
        thumb: `an image`,
        thumbAlt: "",
      },
    };

    const { container } = render(<ProjectInfo {...requiredProps} />);
    expect(container).toMatchSnapshot();
  });
});
