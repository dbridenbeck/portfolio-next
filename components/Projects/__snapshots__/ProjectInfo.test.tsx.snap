// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfo should render without errors 1`] = `
<div>
  <div
    class="ProjectInfo__ProjectInfoContainer-nswi3l-0 ipQHJH"
    style="opacity: 0; height: 0%;"
  >
    <p
      class="ProjectInfo__InfoP-nswi3l-1 hlATVn"
    >
      The client's goal was to limit their reliance on Shopify by having a custom site and use images to tell their story. It was fun implementing a design file into a pixel-perfect responsive site and managing state for checkout. Roadmap is to migrate to Nextjs and add route animations.
      <a
        class="ProjectInfo__ProjectLink-nswi3l-2 ihHhvI"
        href="http://whidbeyherbal.com"
        target="_blank"
      >
        Visit the website 
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right fa-w-14 "
          data-icon="arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
            fill="currentColor"
          />
        </svg>
      </a>
    </p>
  </div>
</div>
`;
